src/Conversion/ONNXToTorch/NN/GemmOp.cpp

```diff
@@ -0,0 +1,159 @@
+    auto iVal = IntegerAttr::get(iType, val);
+    return rewriter.create<ConstantIntOp>(loc, iVal);
+  }
+
+  ONNXGemmOpToTorchLowering(TypeConverter &typeConverter, MLIRContext *ctx)
+      : ConversionPattern(
+            typeConverter, mlir::ONNXGemmOp::getOperationName(), 1, ctx) {}
+
+  LogicalResult matchAndRewrite(Operation *op, ArrayRef<Value> operands,
+      ConversionPatternRewriter &rewriter) const final {
+    ONNXGemmOp gemmOp = llvm::dyn_cast_or_null<ONNXGemmOp>(op);
+    assert(gemmOp && "Expecting op to have a strong type");
+
+    auto loc = gemmOp.getLoc();
+    mlir::MLIRContext *context = gemmOp.getContext();
+
+    auto alpha = gemmOp.alphaAttr();   // ::mlir::FloatAttr
+    auto beta = gemmOp.betaAttr();     // ::mlir::FloatAttr
+    auto transA = gemmOp.transAAttr(); // ::mlir::IntegerAttr
+    auto transB = gemmOp.transBAttr(); // ::mlir::IntegerAttr
+
+    Value f0v = getIntValue(0, rewriter, context, loc);
+    Value f1v = getIntValue(1, rewriter, context, loc);
+
+    auto aTensor = getTorchTensor(gemmOp.A(), rewriter, context, loc);
+    auto bTensor = getTorchTensor(gemmOp.B(), rewriter, context, loc);
+    auto cTensor = getTorchTensor(gemmOp.C(), rewriter, context, loc);
+
+    auto resultType = toTorchType(context, gemmOp.getResult().getType());
+
+    // Transpose A and B. Transpose on Torch is only 2d or less.
+    Value transposeAVal = (transA) ? rewriter.create<AtenTransposeIntOp>(
+                                         loc, resultType, aTensor, f0v, f1v)
+                                   : aTensor;
+    Value transposeBVal = (transB) ? rewriter.create<AtenTransposeIntOp>(
+                                         loc, resultType, bTensor, f0v, f1v)
+                                   : bTensor;
+
+    // Compute Y = alpha * A’ * B’ + beta * C
+    // Scalar multiplication with alpha(alpha * A’)
+    // and beta(beta * C) values.
+    Value alphaMulResult = NULL, betaMulResult = NULL;
+    if (alpha) {
+      Value alpha3v = getFloatValue(alpha, rewriter, loc);
+      alphaMulResult = rewriter.create<AtenMulScalarOp>(
+          loc, resultType, transposeAVal, alpha3v);
+    }
+
+    if (beta) {
+      Value beta3v = getFloatValue(beta, rewriter, loc);
+      betaMulResult =
+          rewriter.create<AtenMulScalarOp>(loc, resultType, cTensor, beta3v);
+    }
+
+    // Bmm Operation ((alpha * A’) * B’)
+    Value bmmValue;
+    if (alphaMulResult)
+      bmmValue = rewriter.create<AtenBmmOp>(
+          loc, resultType, alphaMulResult, transposeBVal);
+    else
+      bmmValue = rewriter.create<AtenBmmOp>(
+          loc, resultType, transposeAVal, transposeBVal);
+
+    // Addition ((alpha * A’ * B’) + (beta * C))
+    Value result;
+    if (betaMulResult)
+      result = rewriter.create<AtenAddTensorOp>(
+          loc, resultType, bmmValue, betaMulResult, f1v);
+    else
+      result = rewriter.create<AtenAddTensorOp>(
+          loc, resultType, bmmValue, transposeBVal, f1v);
```

Comment:
I believe the wrong values is being passed here.
```suggestion
          loc, resultType, bmmValue, cTensor, f1v);
```

Reply:
Thanks! Fixed.